### PR TITLE
Workflow and scope change improvements

### PR DESCRIPTION
**Pull request by Gustav-Eikaas**
# Description

Please include a summary of the change and which issue is fixed.
List any external dependencies that are required for this change.

Completes: AB#FILL_IN_YOUR_ISSUE_ID

## Checklist:

-   [ ] I have performed a self-review of my own code.
-   [ ] I have linked my DevOps task using the AB# tag.
-   [ ] My code is easy to read, and comments are added where needed.
-   [ ] My code is covered by tests.

### CODE REVIEW

File: src/apps/ScopeChangeRequest/Components/Form/Origins/SelectPunch.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+export const SelectPunch = ({ setOriginId }: SelectPunchProps): JSX.Element => {
+    const [isValidPunch, setIsValidPunch] = useState<boolean>(false);
+    const [isChecking, setIsChecking] = useState<boolean>();
+    const [hasChecked, setHasChecked] = useState<boolean>();
+
+    const { procosys } = useHttpClient();
+
+    async function validatePunch(plNumber) {
+        console.log('Validating punch');
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Form/Origins/SelectPunch.tsx:19]: remove

File: src/apps/ScopeChangeRequest/Components/Form/ScopeChangeRequestForm.tsx [hunk 2]
```diff
@@ -29,25 +34,28 @@ interface CreateScopeChangeParams {
     draft: boolean;
 }
 
+export interface Origin {
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Form/ScopeChangeRequestForm.tsx:37]: remove

File: src/apps/ScopeChangeRequest/Components/SearchableDropdown/PCSLink.tsx [hunk 3]
```diff
@@ -174,6 +200,21 @@ export const PCSLink = ({ relatedObjects, setRelatedObjects }: PCSLinkProps): JS
...
+    }
+
+    if (x.type === 'discipline') {
+        return <Icon name="school" color={tokens.colors.interactive.primary__resting.hex} />;
+    }
+
+    if (x.type === 'tag') {
+        return <Icon name="tag" color={tokens.colors.interactive.primary__resting.hex} />;
+    }
+    return <></>;
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/SearchableDropdown/PCSLink.tsx:215]: return null

File: src/apps/ScopeChangeRequest/Components/SearchableDropdown/PCSLink.tsx [hunk 4]
```diff
@@ -184,11 +225,14 @@ const Inline = styled.span`
...
     display: flex;
-    justify-content: space-between;
+    width: fit-content;
     align-items: center;
     font-size: 16px;
     padding: 5px;
+    background-color: ${tokens.colors.ui.background__light.hex};
+    border-radius: 35%;
 `;
+// justify-content: space-between;
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/SearchableDropdown/PCSLink.tsx:235]: remove comment

File: src/apps/ScopeChangeRequest/Components/Upload.tsx [hunk 5]
```diff
@@ -11,7 +11,7 @@ interface UploadProps {
 
 export const Upload = ({ attachments, setAttachments }: UploadProps): JSX.Element => {
     const [rejectedFiles, setRejectedFiles] = useState<FileRejection[]>([]);
-
+    const maxSize = 100 * 1000 ** 2;
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Upload.tsx:14]: maxSizeInBytes

Comment (Gustav-Eikaas) [reply src/apps/ScopeChangeRequest/Components/Upload.tsx:14]: readonly

File: src/apps/ScopeChangeRequest/Components/Workflow/WorkflowCriterias.tsx [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+                                                ? `Signature from ${criteria.value} required.`
+                                                : `Signed by ${criteria.signedBy.firstName} ${criteria.signedBy.lastName}`
+                                        }
+                                    >
+                                        <span>{step.name}</span>
+                                    </Tooltip>
+                                    {criteria.signedAtUtc ? (
+                                        <div
+                                            style={{ fontSize: '14px' }}
+                                        >{`${day}/${month}/${year} ${hour}:${paddedMinutes} - ${criteria.signedBy.firstName} ${criteria.signedBy.lastName} `}</div>
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Workflow/WorkflowCriterias.tsx:47]: convert to function

File: src/apps/ScopeChangeRequest/Components/Workflow/WorkflowCriterias.tsx [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+            })}
+        </>
+    );
+};
+
+const statusFunc = (item: WorkflowStep): WorkflowStatus => {
+    if (item.isCompleted) {
+        return 'Completed';
+    }
+    if (item.isCurrent) {
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Workflow/WorkflowCriterias.tsx:82]: if else

File: src/apps/ScopeChangeRequest/Components/Workflow/WorkflowCriterias.tsx [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+    margin-top: 0.5rem;
+    width: -webkit-fill-available;
+`;
+
+const Inline = styled.span`
+    display: flex;
+    align-items: center;
+`;
+
+const convertUtcToLocalDate = (date: Date) =>
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Workflow/WorkflowCriterias.tsx:129]: Utils

File: src/apps/ScopeChangeRequest/Components/Workflow/WorkflowIcon.tsx [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+            );
+
+        case 'Failed':
+            return (
+                <Icon
+                    name="remove_outlined"
+                    height={'28.8'}
+                    width={'28.8'}
+                    color="grey"
+                // color={tokens.colors.infographic.primary__energy_red_100.hex}
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Workflow/WorkflowIcon.tsx:45]: remove

File: src/packages/Sidesheet/Functions/openSidesheet.ts [hunk 10]
```diff
@@ -6,7 +6,6 @@ import { dispatch, readState } from '../State/actions';
 
 export function openSidesheet<T>(SidesheetContent?: React.FC<T>, props?: T): void {
     if (!SidesheetContent && !props) return;
-
```
Comment (Gustav-Eikaas) [inline src/packages/Sidesheet/Functions/openSidesheet.ts:9]: revert

**Review by jaysencpp — APPROVED** [review]

**Pull request merged**